### PR TITLE
api fix

### PR DESCRIPTION
**Pull request by zekth**
Fixing: https://github.com/denoland/deno_std/issues/308

Sorry for breaking it.

### CODE REVIEW

File: testing/mod.ts [hunk 1]
```diff
@@ -124,8 +124,8 @@ function previousPrinted(name: string, results: TestResults): boolean {
 async function createTestCase(
   stats: TestStats,
   results: TestResults,
-  { fn, name }: TestDefinition,
-  exitOnFail: boolean
+  exitOnFail: boolean,
+  { fn, name }: TestDefinition
```
Comment (ry) [inline testing/mod.ts]: Why didn't any tests fail before? Is there some API that's not being exercised?

cc @chiefbiiko

**Comment by bartlomieju** [comment]
Why did CI pass in #307?

File: testing/mod.ts (hunk 1 above)
Comment (zekth) [reply testing/mod.ts]: As it's not exported, the only way to test it would be to do an assert in `runTests()` for `runTests()`. Maybe chiefbiiko has a clue to test it?

Comment (ry) [reply testing/mod.ts]: Something isn't being exercised - otherwise #307 wouldn't have been green. We should add a test with this commit that was failing before.
Maybe @chiefbiiko can help.

**Comment by zekth** [comment]
Just pushed a proposal for testing runTest in parallel and serial. Ensuring it runs properly and does not throw exception. Had to flush `tests` array in the mod to make it work.

It's a bit sketchy but works. As i mentionned in #307 could be refactored in a further PR. Your thoughts?

File: testing/mod.ts (hunk 1 above)
Comment (chiefbiiko) [reply testing/mod.ts]: @ry @zekth Yea, sorry, I missed adding a test for parallel execution. See patch below.

**Comment by chiefbiiko** [comment]
I suggest applying below patch. It adds the missing parallel tests.

To enable running a test subset in parallel after testing all serially (in `testing/main.ts`) I introduced the `only` and `skip` bits to `RunOptions`. Couldn't use `testing.setFilter` to select a subset bc that filter is applied globally.

@ry @zekth @bartlomieju

```
diff --git a/testing/main.ts b/testing/main.ts
index 077d662..00502b2 100644
--- a/testing/main.ts
+++ b/testing/main.ts
@@ -1,3 +1,11 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
 import { runTests } from "./mod.ts";
-runTests();
+
+async function main() {
+  // Testing entire test suite serially
+  await runTests();
+  // Testing parallel execution on a subset that does not depend on exec order
+  await runTests({ parallel: true, only: /^testing/ });
+}
+
+main();
diff --git a/testing/mod.ts b/testing/mod.ts
index d0d755e..71f27db 100644
--- a/testing/mod.ts
+++ b/testing/mod.ts
@@ -10,7 +10,7 @@ export interface TestDefinition {
 }
 
 let filterRegExp: RegExp | null;
-const tests: TestDefinition[] = [];
+const candidates: TestDefinition[] = [];
 
 let filtered = 0;
 
@@ -35,7 +35,7 @@ export function test(t: TestDefinition | TestFunction): void {
     throw new Error("Test function may not be anonymous");
   }
   if (filter(name)) {
-    tests.push({ fn, name });
+    candidates.push({ fn, name });
   } else {
     filtered++;
   }
@@ -124,8 +124,8 @@ function previousPrinted(name: string, results: TestResults): boolean {
 async function createTestCase(
   stats: TestStats,
   results: TestResults,
-  { fn, name }: TestDefinition,
-  exitOnFail: boolean
+  exitOnFail: boolean,
+  { fn, name }: TestDefinition
 ): Promise<void> {
   const result: TestResult = results.cases.get(results.keys.get(name));
   try {
@@ -197,6 +197,8 @@ async function runTestsSerial(
 export interface RunOptions {
   parallel?: boolean;
   exitOnFail?: boolean;
+  only?: RegExp;
+  skip?: RegExp;
 }
 
 /**
@@ -205,11 +207,16 @@ export interface RunOptions {
  */
 export async function runTests({
   parallel = false,
-  exitOnFail = false
+  exitOnFail = false,
+  only = /[^\s]/,
+  skip = /^\s*$/
 }: RunOptions = {}): Promise<void> {
+  const tests: TestDefinition[] = candidates.filter(
+    ({ name }) => only.test(name) && !skip.test(name)
+  );
   const stats: TestStats = {
     measured: 0,
-    ignored: 0,
+    ignored: candidates.length - tests.length,
     filtered: filtered,
     passed: 0,
     failed: 0
```

File: testing/test.ts [hunk 2]
```diff
@@ -1,255 +1,10 @@
...
-} from "../testing/asserts.ts";
+import { runIfMain } from "./mod.ts";
+
 import "./format_test.ts";
 import "./diff_test.ts";
 import "./pretty_test.ts";
 import "./asserts_test.ts";
 import "./bench_test.ts";
 
-test(function testingAssertEqualActualUncoercable() {
```
Comment (ry) [inline testing/test.ts]: These do need to be moved, but can you do this in a separate PR?

Comment (zekth) [reply testing/test.ts]: Yes. I'll drop this commit.

**Comment by zekth** [comment]
@ry ready to merge.
